ui/app/product-versions/components/pnc-product-version-link/pncProductVersionLink.js

```diff
@@ -18,32 +18,31 @@
 (function () {
   'use strict';
 
-  angular.module('pnc.build-configs').component('pncBuildConfigProductsTab', {
+  angular.module('pnc.product-versions').component('pncProductVersionLink', {
     bindings: {
-      buildConfig: '<',
-      productVersions: '<'
+      /**
+       * object representing SCM Repository
+       */
+      productVersion: '<'
     },
-    require : {
-      mainCtrl: '^^pncBuildConfigDetailMain'
-    },
-    templateUrl: 'build-configs/detail/products-tab/pnc-build-config-products-tab.html',
-    controller: [Controller]
+    templateUrl: 'product-versions/components/pnc-product-version-link/pnc-product-version-link.html',
+    controller: ['ProductVersionResource', Controller]
   });
 
-
-  function Controller() {
+  function Controller(ProductVersionResource) {
     var $ctrl = this;
-
+    $ctrl.productVersionData = $ctrl.productVersion;
     // -- Controller API --
 
-    $ctrl.displayFields = ['name', 'version'];
 
     // --------------------
-
-    $ctrl.$onInit = function() {
-      $ctrl.page = $ctrl.productVersions;
+    $ctrl.$onInit = function () {
+      if($ctrl.productVersion){
```

Comment:
Hi @alexcreasy , this because the productVersion information in the build config object does not contain product information. The BC data usually looks like this:
```
{
      "id": "string",
      "name": "string",
      "description": "string",
      "buildScript": "string",
    
      ...

      "productVersion": {
        "id": "string",
        "version": "string",
        "attributes": {
          "additionalProp1": "string",
          "additionalProp2": "string",
          "additionalProp3": "string"
        }

      ...

      }
    }
```
But for the link we need a full version of `productVersion` info that contains product name and product id so that we can redirect to the right product version page.

Reply:
OK that makes sense... it's poor design on the REST API really, but lets go with it :)